tests/queries/0_stateless/02115_rewrite_local_join_right_distribute_table.sql

```diff
@@ -23,6 +23,10 @@ select t1.* from t1_all t1 join t2_all t2 on t1.a = t2.a ORDER BY t1.a;
 
 SELECT '-';
 
+-- make sure data is fully written when reading from distributed
```

Comment:
But the data is fully written to a MergeTree table, regardless of OPTIMIZE.

Reply:
I'm not sure about my fix, but I want to keep an eye on it's flakiness.